hw/xbox/nv2a/pgraph_texture_replacer.c

```diff
@@ -0,0 +1,26 @@
+#include "pgraph_texture_replacer.h"
+
+
+void texture_download(TextureBinding *texture, SurfaceBinding *surface)
+{
+    void *pixels = surface->texture;
+     char filename[20];
+
+    snprintf(filename, sizeof(filename), "%lu.bmp", surface->vram_addr);
+   
+    SDL_Surface * surf = SDL_CreateRGBSurfaceFrom(pixels, surface->width, surface->height, 8*4, surface->width*4, 0,0,0,0);
+    SDL_SaveBMP(surf, filename);
+    SDL_FreeSurface(surf);
+
+    free(pixels);
+    free(surface->texture);
+    
```

Comment:
I suggest you null-out surface->texture after you free it so it is clear that it's no longer valid.